base_custom_attribute/models/attribute_attribute.py

```diff
@@ -66,7 +66,11 @@ def _build_attribute_field(self, subgroup):
                 if domain:
                     kwargs["domain"] = self.domain
                 else:
-                    ids = [op.value_ref.id for op in self.option_ids]
+                    ids = []
+                    for option in self.option_ids:
+                        # Display only options linked to an existing object
+                        if option.value_ref:
+                            ids.append(option.value_ref.id)
```

Comment:
FYI, you can write it in a comprehension list as well.
```py
ids = [op.value_ref.id for op in self.option_ids if option.value_ref] 
```

Reply:
That's right, I even write it like that in another place. 
Made the correction.